lib/cdo/aws/s3.rb

```diff
@@ -47,7 +47,17 @@ def initialize(message = nil)
     # the credentials specified in the CDO config.
     # @return [Aws::S3::Client]
     def self.connect_v2!
-      self.s3 ||= Aws::S3::Client.new
+      self.s3 ||= if CDO.aws_s3_emulated?
```

Comment:
An interesting finding this morning...TIL we have a second S3 client for "middleware" (files/sources/etc):

https://github.com/code-dot-org/code-dot-org/blob/093b289ffc52a81d1e69c83610d776da9436fbeb/dashboard/legacy/middleware/helpers/bucket_helper.rb#L18

S3 emulation seems to be working though, even without the ternary like we've got here. [I think that makes sense given the hierarchy of credentials listed here](https://rubydoc.info/gems/aws-sdk-s3/1.98.0/Aws%2FS3%2FClient:initialize). Do you think it makes sense to put this ternary in place everywhere we use S3, or rely on the defaults to do the work for us everywhere?

Reply:
Oh nvm, got turned around. This uses the same S3 client as the one where your changes are, so we're good!